src/mbgl/gl/attribute.hpp

```diff
@@ -242,7 +247,18 @@ class Attributes {
     static constexpr std::size_t Index = TypeIndex<A, As...>::value;
 
     static Locations bindLocations(const ProgramID& id) {
-        return Locations { bindAttributeLocation(id, Index<As>, As::name())... };
+        std::set<std::string> activeAttributes;
+
+        int32_t attributeCount = getActiveAttributeCount(id);
+        int32_t maxAttributeLength = getMaxAttributeNameLength(id);
+
+        for (int32_t i = 0; i < attributeCount; i++) {
+            std::string attributeName = getAttributeName(id, maxAttributeLength, i);
+            activeAttributes.emplace(attributeName);
+        }
+
+        return Locations{ bindAttributeLocation(
+            id, locationToBindAttribute(activeAttributes, As::name()), As::name())... };
```

Comment:
This is close, but we want to number them in the order defined by the `As` argument pack, rather than lexicographical order by name. The latter might put an attribute that's sometimes disabled at index 0, which as we know, causes issues. Try this:

```C++
AttributeLocation location = 0;
return Locations {
    (activeAttributes.count(As::name()) ? bindAttributeLocation(id, location++, As::name()) : -1)...
};
```